sdk/serialization/azure-json/src/main/java/com/azure/json/implementation/DefaultJsonReader.java

```diff
@@ -79,17 +80,20 @@ public static JsonReader fromReader(Reader reader, JsonOptions options) throws I
 
     private DefaultJsonReader(JsonParser parser, boolean resetSupported, byte[] jsonBytes, String jsonString,
         JsonOptions options) {
-        this(parser, resetSupported, jsonBytes, jsonString, options.isNonNumericNumbersSupported());
+        this(parser, resetSupported, jsonBytes, jsonString, options.isNonNumericNumbersSupported(),
+            options.isJsoncSupported());
     }
 
     private DefaultJsonReader(JsonParser parser, boolean resetSupported, byte[] jsonBytes, String jsonString,
-        boolean nonNumericNumbersSupported) {
+        boolean nonNumericNumbersSupported, boolean jsoncSupported) {
         this.parser = parser;
-        this.parser.configure(JsonReadFeature.ALLOW_NON_NUMERIC_NUMBERS.mappedFeature(), nonNumericNumbersSupported);
         this.resetSupported = resetSupported;
+        this.parser.configure(JsonReadFeature.ALLOW_NON_NUMERIC_NUMBERS.mappedFeature(), nonNumericNumbersSupported);
+        this.parser.configure(JsonParser.Feature.ALLOW_COMMENTS, jsoncSupported);
```

Comment:
Jackson breaks out comment processing into a few different feature settings, do we want to enable the others as well when jsoncSupported is enabled?

Reply:
Also, since this is a generic feature we'll need to update code in azure-core-serialization-json-gson, azure-core-serialization-json-jackson, and azure-core

Reply:
Discussed offline - we'll do these on demand as folks need them.

Reply:
> Also, since this is a generic feature we'll need to update code in azure-core-serialization-json-gson, azure-core-serialization-json-jackson, and azure-core

Will do this in a separate PR.